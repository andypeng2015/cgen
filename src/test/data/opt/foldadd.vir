module foo

export function w $foo(w %x) {
@start:
  %y = add.w %x, 1_w
  %z = add.w %x, 1_w
  %r = add.w %y, %z
  ret %r
}
