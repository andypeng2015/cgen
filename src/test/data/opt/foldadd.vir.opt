module foo

export function w $foo(w %x) {
@2:
  %2 = add.w %x, 0x2_w ; @8
  %3 = add.w %x, %2 ; @9
  ret %3
}
