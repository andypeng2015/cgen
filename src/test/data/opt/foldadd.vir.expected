module foo

export function w $foo(w %x) {
@0:
  %0 = add.w %x, 0x2_w ; @4
  %1 = add.w %x, %0 ; @5
  ret %1
}
